chapters/develop-organize/drills/reading/tasks.md

```diff
@@ -1,12 +1,95 @@
 # Tasks
 
-Tasks are types of drills that require a solution.
-This is opposition to [questions](questions.md) that require an answer.
+Tasks are types of drills that require a **detailed solution**.
+This is opposition to [questions](questions.md) that require a **simple answer**.
 
-TODO: goals
+Tasks verify a wider range of information than quizzes and have the potential to go into more depth regarding a particular subject.
+A task may come in the form of an essay, a report, a formal proof, a source code file, etc.
+This comes with the disadvantage that grading some of these tasks is not easily automated.
 
-TODO: directory structure
+While a code snippet may be easily graded automatically via tests, it is impossible to do so for an essay for example.
 
-TODO: examples
+## Purpose
 
-TODO: guides for designing tasks
+Tasks are intended to be forms of self-study or formal examination.
+They provide a means to evaluate learner understanding.
+This creates a positive feedback loop between trainers and trainees whereby both can track the learners' progress in real time.
+
+Furthermore, tasks serve as practice items for learners to solve at their own pace, in conjunction with questions.
+They may also be used for evaluation items that contribute to the learners' overall grade.
+
+Another desired aspect is that tasks be easily extractable and convertible between formats, as described in the ["Deployment"](#deployment) section, and repurposed by other trainers.
+
+## Steps to Develop a Task
+
+To create a new task from scratch, a educator should:
+
+1. Brainstorm the question.
+
+1. Write it down using the Open Education Hub format proposed in this document.
+
+1. Push it to the repository.
```

Comment:
That's how you get "Add files via upload" commits. We should maybe have a separate section on git and GitHub, or advertise a tutorial, stating that being knowledgeable about git is a necessary prerequisite. Without that one could not apply our methodology, since it revolves around git.

Reply:
You are right, it is a prerequisite so it is not in the scope of methodology. At most, we can provide links to get started with git / github.